### ISSUE TITLE
Compiling GRASP using gfortran-10 fails

### ISSUE THREAD

**Issue by jagot** [compas/grasp#46]
Compiling with the latest version of gfortran, i.e. 10, fails with the following error:
```
/tmp/grasp/src/lib/lib9290/iniest2.f90:82:23:

   79 |          CALL DCOPY (NS, VEC(NS*(J-1)+1), 1, BASIS(NCF*(J-1)+1), 1)
      |                         2
......
   82 |       CALL DCOPY (NIV, EIGVAL, 1, BASIS(NIV*NCF+1), 1)
      |                       1
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
make[2]: *** [src/lib/lib9290/CMakeFiles/9290.dir/iniest2.f90.o] Error 1
make[1]: *** [src/lib/lib9290/CMakeFiles/9290.dir/all] Error 2
make: *** [all] Error 2
```
It seems it's the same problem as https://github.com/scipy/scipy/issues/11611

Compiling with version 9 works, i.e. on a Mac where I have multiple versions installed, I can do
```sh
> FC=gfortran-9 ./configure.sh
> cd build && make
```

**Comment by mortenpi** [comment]
Thanks for the report and the ref. It might be a while before we fix the underlying issues.

But based on the [GCC porting guide](https://gcc.gnu.org/gcc-10/porting_to.html) it should be sufficient to drop something like this into the `CMakeLists.user` file:

```cmake
set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch")
```

to work around the error. Or equivalent into `Make.user`, for those using `Makefile`s.

**Comment by jongrumer** [comment]
Great, I just encountered this as well, and hadn't noticed your issue so created a new one - stupid. Anyway, I found the same workaround as Morten. We have tons of these missmatches around the code unfortunately, both scalar-rank-1 and real4-real8 etc. Not sure what to do really, I guess we could add the `-fallow-argument-mismatch` flag by default, but has to come with an if-statement since it only works for `gfortran-10` and not below.
